### PR TITLE
openrc-init: ignore an empty string in argv[1]

### PR DESCRIPTION
**Pull request by floppym**
Bug: https://bugs.gentoo.org/803536

### CODE REVIEW

**Comment by vapier** [comment]
but like, why ?  it seems like an error to run `init ""`.

**Comment by floppym** [comment]
> but like, why ? it seems like an error to run init "".

I agree, but it was requested in https://bugs.gentoo.org/803536#c4.

It's worth noting that sysvinit just ignores any argument it doesn't recognize, including an empty string.

**Comment by floppym** [comment]
openrc-init behaves a bit differently in that it treats argv[1] as the default runlevel, and performs no validation.

So maybe a better solution would be to check if the specified runlevel exists instead of checking for an empty string specifically.

**Comment by williamh** [comment]
I'll work on this, but I think a better solution even would be to fall back to the default runlevel if the passed in runlevel doesn't exist.

**Comment by williamh** [comment]
@vapier  @floppym  What do you think?

**Comment by vapier** [comment]
> I agree, but it was requested in https://bugs.gentoo.org/803536#c4.

that doesn't make it a good idea ... people request all sorts of things.
commit messages should also have complete explanation & logic in them rather than a loose link to a website.

> I'll work on this, but I think a better solution even would be to fall back to the default runlevel if the passed in runlevel doesn't exist.

if we're running as pid 1, that seems reasonable -- emit a warning, and fallback to the default.

if we aren't running as pid 1, we should show an error & exit non-zero.

**Comment by floppym** [comment]
> commit messages should also have complete explanation & logic in them rather than a loose link to a website.

The reason for the change seems obvious to me. If William wanted some further explanation, I assume he would have asked for it.

**Comment by floppym** [comment]
> that doesn't make it a good idea ... people request all sorts of things.

Let me rephrase: I agree it is an error to invoke openrc-init with an empty string as the first argument. However the request to guard against such a user error also seems reasonable to me.

**Comment by williamh** [comment]
@floppym  I need to check rc_runlevel_exists. The bug you linked says that an empty string passed to that function returns true. If that's the case, that seems to be the underlying issue.
